frontend/src/components/question-creation/question-elements/QuestionTextElement.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useContext } from "react";
+import update from "immutability-helper";
+import { Flex, Text, Textarea } from "@chakra-ui/react";
+import ResizeTextarea from "react-textarea-autosize";
+
+import QuestionEditorContext from "../../../contexts/QuestionEditorContext";
+import { QuestionElementType } from "../../../types/QuestionTypes";
+
+interface QuestionTextElementProps {
+  id: string;
+  data: string;
+}
+
+const QuestionTextElement = ({
+  id,
+  data,
+}: QuestionTextElementProps): React.ReactElement => {
+  const { questionElements, setQuestionElements } = useContext(
+    QuestionEditorContext,
+  );
+
+  const questionCount = questionElements.filter(
+    (element) => element.type === QuestionElementType.QUESTION,
+  ).length;
+  const questionLetter = String.fromCharCode(
+    "a".charCodeAt(0) +
+      (questionElements
+        .filter((element) => element.type === QuestionElementType.QUESTION)
+        .findIndex((element) => element.id === id) %
+        26),
+  );
+
+  const updateQuestionElement = (updatedQuestion: string) => {
```

Comment:
if this is going to be reused for each component, should we also factor it out? and have it accept an error message as input

i'm not sure what the best practice is here, bc if we are defining this function in `QuestionElementItem.tsx` and down passing as props, that defeats the whole context thing? but then having 6 copies of this function also seems wrong

Reply:
+1! i'll refactor it out. I agree, I don't think I should be passing context down as props, so I think I might put it in `QuestionUtils`?

Reply:
i moved the logic into `QuestionUtils` but left the setter in the individual elements! let me know if that's better

Reply:
this is good !!